### PR TITLE
Add soaring-upcoming page.

### PR DESCRIPTION
**Pull request by samuelctabor**
This replaces https://github.com/ArduPilot/ardupilot_wiki/pull/3013 in favour of having a separate version of the soaring page ("soaring-upcoming") that is relevant for the latest features. This reduces clutter on the "stable" page while making the "master" documentation accessible.

The two pages are currently in-sync and I will apply any updates to them as needed to keep them consistent with each other and/or the relevant code. Right now the "master" page is only linked from the "stable" one.

### CODE REVIEW

**Comment by Hwurzburg** [comment]
@samuelctabor not a review, will do that tomorrow....just skimmed....I think you lost SOAR_ENABLE in the reorg...also headings need the underline char to be the same length as the heading....several spots...

**Comment by samuelctabor** [comment]
Thanks, fixed the underlines and a duplicate label.

Re the naming, when a new release is made I'll move the included features from `soaring-upcoming.rst` to `soaring.rst`. So `soaring-upcoming.rst` will always include the features not yet in a stable release.

**Review by Hwurzburg — COMMENTED** [review]
need to add the <soaring-upcoming> page to toc in common-master-features.rst

File: plane/source/docs/soaring-upcoming.rst [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+*S. Tabor, I. Guilliard, A. Kolobov.* `ArduSoar: an Open-Source Thermalling Controller for Resource-Constrained Autopilots <https://arxiv.org/abs/1802.08215/>`_. *International Conference on Intelligent Robots and Systems (IROS), 2018.*
+
+
+.. image:: ../../../images/thermalling.jpg
+
+This picture shows the different phases of flight when using the soaring
+functionality:
+
+#. If modes AUTO, FBWB or CRUISE are entered, and Soaring is enabled, the throttle is set to zero provided the aircraft is above :ref:`SOAR_ALT_MIN<SOAR_ALT_MIN>` altitude and the aircraft then begins gliding.
+#. If the aircraft descends to :ref:`SOAR_ALT_MIN<SOAR_ALT_MIN>` altitude, throttle is re-enabled and the aircraft will begin to climb to the altitude of the next waypoint, in AUTO, or to :ref:SOAR_ALT_CUTOFF<SOAR_ALT_CUTOFF> altitude in CRUISE or FBWB. In AUTO, if the waypoint altitude is less than :ref:`SOAR_ALT_CUTOFF<SOAR_ALT_CUTOFF>` altitude, then Soaring can not begin before reaching the waypoint. If it is above :ref:`SOAR_ALT_CUTOFF<SOAR_ALT_CUTOFF>` altitude, then Soaring can occur once that altitude has been reached.
```
[the remark this answers is not in the document]
Comment (samuelctabor) [reply plane/source/docs/soaring-upcoming.rst]: Thanks, fixed!

**Comment by Hwurzburg** [comment]
I will need to do a few cleanups elsewhere, but merging now...thanks!

**Pull request merged**